libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/src/azure/iot/az_iot_adu_ota.c

```diff
@@ -350,7 +342,11 @@ AZ_NODISCARD az_result az_iot_adu_ota_parse_service_properties(
 
                     _az_RETURN_IF_FAILED(az_json_token_get_string(
                         &jr->token, (char*)az_span_ptr(update_request->workflow.id),
-                        az_span_size(update_request->workflow.id), NULL));
+                        az_span_size(update_request->workflow.id), &out_length));
+
+                    // TODO: find a way to get rid of az_json_token_get_string (which adds a \0 at the end!!!!!!)
+                    //       Preferably have a function that does not copy anything.
+                    update_request->workflow.id = az_span_slice(update_request->workflow.id, 0, out_length);
```

Comment:
is it really a problem that it appends a `NULL` character? Just out of curiosity. We could always create a span around the returned char* and do a `size - 1`. Or is it that doing that is clunky?

Reply:
My plan is to change this function to not use the az_token_get_string, only assign the token slice (and avoid copies). So to maintain the same public contract of no extra '\0's, I trimmed them here.